src/actions/index.js

```diff
@@ -1,83 +1,76 @@
 import {
   firebase,
   applicationsRef,
-  firestore,
   rsvpRef,
 } from '../config/firebase';
 
+const API_PATH = 'https://slohacks-backend-api.herokuapp.com';
+
 export const signUp = (values, callback) => (dispatch) => {
   dispatch({ type: types.SIGN_UP_ATTEMPT });
-  firebase.auth().createUserWithEmailAndPassword(values.email, values.password)
-    .then((userCredential) => {
-      firebase.auth().currentUser.sendEmailVerification()
-        .then(() => {
-          dispatch({ type: types.SIGN_UP_GUCCI, userCredential });
-          callback();
-        }).catch(error => dispatch({ type: types.SIGN_UP_FAIL, error }));
-    }).catch((error) => {
-      dispatch({ type: types.SIGN_UP_FAIL, error });
+  axios
+    .post(`${API_PATH}/users/signup`, {
+      email: values.email,
+      password: values.password,
+    })
+    .then((response) => {
+      const userCredential = response.data;
+      dispatch({ type: types.SIGN_UP_GUCCI, userCredential });
+      callback();
+    })
+    .catch((error) => {
+      const { errorMessage } = error.response.data;
+      dispatch({ type: types.SIGN_UP_FAIL, errorMessage });
     });
 };
 
 export const login = values => (dispatch) => {
   dispatch({ type: types.LOGIN_ATTEMPT });
-  firebase.auth().signInWithEmailAndPassword(values.email, values.password)
-    .then((userCredential) => {
-      const {
-        user: { emailVerified, uid },
-      } = userCredential;
-      if (emailVerified) {
-        const appRef = firestore.collection('applications').doc(`${uid}`);
-        const attendRef = firestore.collection('rsvp').doc(`${uid}`);
-        appRef.get().then((app) => {
-          if (app.exists) {
-            attendRef.get().then((rsvp) => {
-              const appData = app.data();
-              if (!rsvp.exists && appData.status === 1) {
-                dispatch({ type: types.UPDATE_APPLICATION_TRUE, app: appData, rsvpInv: false });
-                dispatch({ type: types.LOGIN_GUCCI, userCredential });
-              } else {
-                dispatch({ type: types.UPDATE_APPLICATION_TRUE, app: appData, rsvpInv: true });
-                dispatch({ type: types.LOGIN_GUCCI, userCredential });
-              }
-            });
-          } else {
-            dispatch({ type: types.UPDATE_APPLICATION_FALSE });
-            dispatch({ type: types.LOGIN_GUCCI, userCredential });
-          }
-        }).catch(() => {
-          dispatch({ type: types.UPDATE_APPLICATION_FALSE });
-          dispatch({ type: types.LOGIN_GUCCI, userCredential });
-        });
-      } else {
-        dispatch({ type: types.LOGIN_FAIL, error: { message: 'Email not verified, please verify your email.' } });
-      }
-    }).catch((error) => {
-      dispatch({ type: types.LOGIN_FAIL, error });
+  axios
+    .post(`${API_PATH}/users/login`, {
+      email: values.email,
+      password: values.password,
+    })
+    .then((response) => {
+      const userCredential = response.data;
```

Comment:
console log the usercredential object, there is also a token we need to store. Look at auth reducer and see how we can improve the global state